rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+---
+authors: Grzegorz Zdunek (grzegorz.zdunek@goteleport.com)
+state: draft
+ ---
+
+# RFD 97 - Teleport Connect usage metrics
+
+## Required Approvers
+
+* Engineering:? @ravicious @zmb3
+* Product: @klizhentas @xinding33
+
+## What
+
+Collect downloads and usage metrics of Teleport Connect.
+
+## Why
+
+Currently, the team has no information on how many users download, install and use Teleport Connect on a daily basis.
+In order to effectively plan the development of the product, the team should also know what is the adoption of new
+features, which ones are the most popular and which are problematic for users.
+
+## Details
+
+Connect will have a stateless metrics service that will send events to an unauthorized gRPC endpoint, exposed by TSH
+daemon. The anonymization should be done in `tshd` (to be open on how it is done and what is sent further). Events will
+be sent to `prehog` directly.
+It was considered to use authorized endpoint provided by cluster's `Auth Service`, but it seems to not work well for
+Connect for a few reasons:
+
+- Some events do not belong to any cluster, like _app start_ event.
+- A user can use Connect while not logged in. For example, new tabs can be opened after the session expires.
+
+Since Connect will use endpoint that is not aware of the cluster context, it must provide some value that will
+differentiate events coming from multiple application instances. To do this, each event will have `distinct_id` field.
+It will be supplied with UUID generated by Connect, which will be generated once and stored in a file in the _App Data_
+directory, so it won't change between restarts.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, no data will be collected.
```

Comment:
A question a bit out of the scope of this RFD but since it has to do with collection of data:
Is there a difference between some data that will _always_ be collected (like a Connect user agent when connecting to a cluster or something) and the data that will be opted out of collecting? How are the other teams handling this. Does `tsh` and the web have it's own opt in/out flow? I'd like to compare.

Reply:
Good question, @klizhentas will `web` ask a user if they agree to send the data?

Reply:
Let's push this question out of scope for now. For Teleport connect, users will always have to acknowledge that behavioral data will be submitted.